src/awkward1/highlevel.py

```diff
@@ -406,16 +406,11 @@ def _str(self, limit_value=85):
             return self._array._str(limit_value=limit_value)
 
         def _repr(self, limit_value=40, limit_total=85):
-            import awkward1.operations.structure
-
             suffix = _suffix(self)
             limit_value -= len(suffix)
 
-            layout = awkward1.operations.structure.with_cache(
-                self._layout, {}, chain="last", highlevel=False
-            )
             value = awkward1._util.minimally_touching_string(
-                limit_value, layout, self._behavior
+                limit_value, self._layout, self._behavior
```

Comment:
p.s. I think this `Mask` object has some bugs: it doesn't have `self._layout`, but its parent array would, right?

Reply:
That's entirely likely. I think most of the examples with mask immediately call `__getitem__` in it, so bugs in its `__repr__` might not have been caught.